src/resolve/resolved-dns-query.c

```diff
@@ -323,11 +343,32 @@ void dns_query_candidate_notify(DnsQueryCandidate *c) {
 
         state = dns_query_candidate_state(c);
 
-        if (DNS_TRANSACTION_IS_LIVE(state))
+        if (DNS_TRANSACTION_IS_LIVE(state)) {
+                DnsTransaction *t;
+                bool at_least_one_succeeded = false;
+                SET_FOREACH(t, c->transactions) {
+                        if (t->state == DNS_TRANSACTION_SUCCESS) {
+                                at_least_one_succeeded = true;
+                                break;
+                        }
+                }
+                if (at_least_one_succeeded && !c->timeout_event_source) {
+                        event_reset_time_relative(
+                                        c->query->manager->event,
+                                        &c->timeout_event_source,
+                                        CLOCK_BOOTTIME,
+                                        SD_RESOLVED_CANDIDATE_TIMEOUT_USEC,
+                                        0, on_candidate_timeout, c,
+                                        0, "candidate-timeout", true);
+                }
```

Comment:
no {} around single line if blocks.

it's fine to ignore errors here, but please cast the result to (void) to clarify you are doing this intentionally.

our current coding style suggests to prefix parameters in function calls which are passed as "opaque" values (i.e. 0, NULL and so on), with the parameter name in comment. i.e. the 5th and 8th param should be prefixed like that. (i.e. write `/* accuracy_usec= */` and `/* priority= */`

Reply:
actually, please drop the check for `!c->timeout_event_source` from the `if` expression, and instead pass the last argument as `false`, which has the same effect